src/vs/workbench/api/browser/mainThreadCustomEditors.ts

```diff
@@ -393,14 +393,21 @@ class MainThreadCustomEditorModel extends ResourceWorkingCopy implements ICustom
 					return;
 				}
 
+				const reason = e.auto
+					? localize('vetoAutoExtHostRestart', "One of the opened editors is a custom editor.")
+					: localize('vetoExtHostRestart', "Custom editor '{0}' could not be saved.", this.resource.path);
+
 				e.veto((async () => {
-					const didSave = await this.save();
-					if (!didSave) {
-						// Veto
+					if (e.auto) {
 						return true;
 					}
-					return false; // Don't veto
-				})(), localize('vetoExtHostRestart', "Custom editor '{0}' could not be saved.", this.name));
+
+					const didSave = await this.save();
+					if (didSave) {
+						return false;  // Don't veto
+					}
+					return true; // Veto
+				})(), reason);
```

Comment:
After thinking about it, I would suggest to veto always if there is a custom editor. Reason being, restarting extensions can be disrupting for custom editors because the extension owning the custom editor may or may not be available and the custom editor will be in a broken state.

Reply:
That's an option. As currently custom editors of both kinds become orphaned on restart. I will change it to do so for now.

But note that I made a followup PR which tries to re-establish/reload in-place the custom editor on restart from both deliberate and involuntary (crash) restart of the extension host. Vetoing only effects deliberate restarts, so even with this change, you still end up with orphaned editors on extension host crash, as described in the linked issue I opened.